src/Compiler/Compiler.cpp

```diff
@@ -376,12 +376,12 @@ bool Compiler::RegisterCommands(TclInterpreter* interp, bool batchMode) {
     }
     std::string actualType = "VERILOG_2001";
     Design::Language language = Design::Language::VERILOG_2001;
-    const std::string file = argv[1];
-    if (strstr(file.c_str(), ".vhd")) {
+    auto file = QString(argv[1]).toLower();
```

Comment:
Compiler must use std::string only. This is an agreement. I suggest to write some function **StringUtils::toLower**

Reply:
fixed. also added unit tests for new utility functions